From 8c0e1a7b8a735fa7fb50cec8492f88b36ecb6053 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@pilppa.org>
Date: Fri, 10 May 2024 10:32:33 -0700
Subject: [PATCH 3/7] LoadHIP force ROCM detection and patches

- set HIP_ROOT_DIR to ROCM_PATH which is set
  by the build scripts
- set ROCM_VERSION

Signed-off-by: Mika Laitio <lamikr@pilppa.org>
---
 cmake/public/LoadHIP.cmake | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/cmake/public/LoadHIP.cmake b/cmake/public/LoadHIP.cmake
index fa39156031f..d1e8f534455 100644
--- a/cmake/public/LoadHIP.cmake
+++ b/cmake/public/LoadHIP.cmake
@@ -41,6 +41,7 @@ endmacro()
 find_package_and_print_version(HIP 1.0)
 
 if(HIP_FOUND)
+  #message( FATAL_ERROR "LoadHIP debug trace1.")
   set(PYTORCH_FOUND_HIP TRUE)
   set(FOUND_ROCM_VERSION_H FALSE)
 
@@ -54,13 +55,16 @@ if(HIP_FOUND)
     file(WRITE ${file} ""
       "#include <rocm_version.h>\n"
       )
+      #message( FATAL_ERROR "LoadHIP debug trace2")
   elseif(EXISTS ${ROCM_INCLUDE_DIRS}/rocm-core/rocm_version.h)
     set(FOUND_ROCM_VERSION_H TRUE)
     file(WRITE ${file} ""
       "#include <rocm-core/rocm_version.h>\n"
       )
+      #message( FATAL_ERROR "LoadHIP debug trace3.")
   else()
     message("********************* rocm_version.h couldnt be found ******************\n")
+    #message( FATAL_ERROR "LoadHIP debug trace4.")
   endif()
 
   if(FOUND_ROCM_VERSION_H)
@@ -90,9 +94,12 @@ if(HIP_FOUND)
     message(STATUS "Caffe2: Header version is: " ${rocm_version_from_header})
     set(ROCM_VERSION_DEV_RAW ${rocm_version_from_header})
     message("\n***** ROCm version from rocm_version.h ****\n")
+  else()
+    message( FATAL_ERROR "LoadHIP FOUND_ROCM_VERSION_H failed, ROCM_PATH: ${ROCM_PATH}")
   endif()
 
   string(REGEX MATCH "^([0-9]+)\.([0-9]+)\.([0-9]+).*$" ROCM_VERSION_DEV_MATCH ${ROCM_VERSION_DEV_RAW})
+  #message( FATAL_ERROR "ROCM_VERSION_DEV_MATCH: ${ROCM_VERSION_DEV_MATCH}, ROCM_VERSION_DEV_RAW: ${ROCM_VERSION_DEV_RAW}, ROCM_VERSION: ${ROCM_VERSION}")
 
   if(ROCM_VERSION_DEV_MATCH)
     set(ROCM_VERSION_DEV_MAJOR ${CMAKE_MATCH_1})
@@ -100,6 +107,8 @@ if(HIP_FOUND)
     set(ROCM_VERSION_DEV_PATCH ${CMAKE_MATCH_3})
     set(ROCM_VERSION_DEV "${ROCM_VERSION_DEV_MAJOR}.${ROCM_VERSION_DEV_MINOR}.${ROCM_VERSION_DEV_PATCH}")
     math(EXPR ROCM_VERSION_DEV_INT "(${ROCM_VERSION_DEV_MAJOR}*10000) + (${ROCM_VERSION_DEV_MINOR}*100) + ${ROCM_VERSION_DEV_PATCH}")
+    set(ROCM_VERSION ${ROCM_VERSION_DEV_INT})
+    set(HIP_ROOT_DIR ${ROCM_PATH})
   endif()
 
   message("ROCM_VERSION_DEV: ${ROCM_VERSION_DEV}")
@@ -112,6 +121,7 @@ if(HIP_FOUND)
   message("HIP_VERSION_MAJOR: ${HIP_VERSION_MAJOR}")
   message("HIP_VERSION_MINOR: ${HIP_VERSION_MINOR}")
   message("TORCH_HIP_VERSION: ${TORCH_HIP_VERSION}")
+  message("USE_FLASH_ATTENTION: ${USE_FLASH_ATTENTION}")
 
   message("\n***** Library versions from dpkg *****\n")
   execute_process(COMMAND dpkg -l COMMAND grep rocm-dev COMMAND awk "{print $2 \" VERSION: \" $3}")
-- 
2.45.2

